packages/story-editor/src/components/videoTrim/test/videoTrimmer.js

```diff
@@ -0,0 +1,309 @@
+  const getCancel = () => screen.getByRole('button', { name: 'Cancel' });
+  const getSubmit = () => screen.getByRole('button', { name: 'Trim' });
+  const rerender = ({ layout: reLayout, trim: reTrim }) => {
+    const reLayoutCtx = {
+      ...layoutCtx,
+      state: { ...layoutCtx.state, ...reLayout },
+    };
+    const reTrimCtx = { ...trimCtx, state: { ...trimCtx.state, ...reTrim } };
+    view.rerender(
+      <LayoutContext.Provider value={reLayoutCtx}>
+        <VideoTrimContext.Provider value={reTrimCtx}>
+          <VideoTrimmer />
+        </VideoTrimContext.Provider>
+      </LayoutContext.Provider>
+    );
+  };
+  return {
+    ...view,
+    getVideoTrimmer,
+    getStartHandle,
+    getEndHandle,
+    getCancel,
+    getSubmit,
+    rerender,
+    layout: layoutCtx.state,
+    trim: trimCtx.state,
+    actions: trimCtx.actions,
+  };
+}
+
+describe('Video Trimmer', () => {
+  describe('as it loads', () => {
+    it('should render empty trimmer if missing page width', () => {
+      const { getVideoTrimmer } = setup({
+        layout: { pageWidth: null },
+      });
+
+      const videoTrimmer = getVideoTrimmer();
+      expect(videoTrimmer).toBeInTheDocument();
+      expect(videoTrimmer).toBeEmptyDOMElement();
+    });
+
+    it('should render empty trimmer if no video length', () => {
+      const { getVideoTrimmer } = setup({
+        trim: { maxOffset: null },
+      });
+
+      const videoTrimmer = getVideoTrimmer();
+      expect(videoTrimmer).toBeInTheDocument();
+      expect(videoTrimmer).toBeEmptyDOMElement();
+    });
+
+    it('should render empty trimmer if no video data', () => {
+      const { getVideoTrimmer } = setup({
+        trim: { videoData: null },
+      });
+
+      const videoTrimmer = getVideoTrimmer();
+      expect(videoTrimmer).toBeInTheDocument();
+      expect(videoTrimmer).toBeEmptyDOMElement();
+    });
+
+    it('should render correctly if all data ready', () => {
+      const { getVideoTrimmer } = setup();
+
+      const videoTrimmer = getVideoTrimmer();
+      expect(videoTrimmer).toBeInTheDocument();
+      expect(videoTrimmer).not.toBeEmptyDOMElement();
+    });
+
+    it('should display initial handles correctly if new trim', () => {
+      const { getStartHandle, getEndHandle } = setup();
+
+      expect(getStartHandle()).toHaveStyle('left: 0.00px');
+      expect(getEndHandle()).toHaveStyle('left: 100.00px');
+    });
+
+    it('should display initial handles correctly if re-trimming', () => {
+      const { getStartHandle, getEndHandle } = setup({
+        trim: { startOffset: 1000, endOffset: 9000 },
+      });
+
+      expect(getStartHandle()).toHaveStyle('left: 10.00px');
+      expect(getEndHandle()).toHaveStyle('left: 90.00px');
+    });
+
+    it('should have enabled cancel button', () => {
+      const { getCancel } = setup();
+
+      expect(getCancel()).toBeEnabled();
+    });
+
+    it('should have enabled trim button only once changed', () => {
```

Comment:
wording is a bit tricky to understand what this is testing

Reply:
Fixed in 73cfc68c93.